#58 - Bug On Heroku:  undefine method "address" for NilClass

ParthivPatel-BTC opened the pull request.

ParthivPatel-BTC committed:
#58 - fixed issue undefine method "address" for NilClass

amitpatelx commented on app/controllers/invoices_controller.rb:
```diff
@@ -124,4 +125,9 @@ def invoice_params
   end
+
+  def is_client_exists?
+    redirect_to clients_path if !Client.exists?
```
use `unless` instead `if !`.

Also the condition is wrong. It should be `current_user.clients.exists?`

amitpatelx commented on app/controllers/invoices_controller.rb:
```diff
@@ -124,4 +125,9 @@ def invoice_params
+
+  def is_client_exists?
+    redirect_to clients_path if !Client.exists?
+    flash[:error] = t('clients.messages.create_client_for_invoice')
```
Flash message should be set before redirect.

amitpatelx commented on app/views/layouts/_left_sidebar.html.haml:
```diff
@@ -8,7 +7,7 @@
               %i.fa.fa-dashboard
               %span Home
-          - if current_user.company.present?
+          - if current_user.company.present? && Client.exists?
```
`Client.exists?` condition is wrong. It should be current_user.clients.exists?

amitpatelx commented on config/locales/en.yml:
```diff
@@ -21,6 +21,9 @@
 
 en:
   hello: "Hello world"
```
Remove unused key

ParthivPatel-BTC committed:
#58 - Changed code as per commented by amit

amitpatelx commented on app/controllers/invoices_controller.rb:
```diff
@@ -1,10 +1,11 @@
   before_action :authenticate_user!
   load_and_authorize_resource
-  before_action :replace_date_params , only: [:create,:update]
+  before_action :replace_date_params, only: [ :create, :update ]
```
Don't give space after/before `[]`. We add spaces in `{}` only

amitpatelx commented on app/controllers/invoices_controller.rb:
```diff
@@ -124,4 +125,9 @@ def invoice_params
         line_items_attributes: [:item_id, :line_total, :price, :quantity, '_destroy', :id, :description])
   end
+
+  def is_client_exists?
```
rename to `require_client!`. `!` is used because you are changing the state of request ie. redirect

amitpatelx commented on app/controllers/invoices_controller.rb:
```diff
@@ -124,4 +125,9 @@ def invoice_params
+
+  def is_client_exists?
+    flash[:error] = t('clients.messages.create_client_for_invoice')
+    redirect_to clients_path unless Client.exists?
```
Shouldn't we check for current user/companies clients only?

ParthivPatel-BTC committed:
#58 - Changed code as per following comments.
1. Don't give space after/before []. We add spaces in {} only
2. Rename to require_client!. ! is used because you are changing the state of request ie. redirect
3. Shouldn't we check for current user/companies clients only?

AmitPatel-BoTreeConsulting commented:
:+1:
